Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/Lib15EF4C78.galaxy

```diff
@@ -247,6 +247,26 @@ void lib15EF4C78_InitCustomScript () {
 
 // Presets
 // Functions
+void lib15EF4C78_gf_AP_ApplyDetectionUpgrades () {
+    // Variable Declarations
+    int lv_player_index;
+
+    // Automatic Variable Declarations
+    const int autoAF3B14EE_ae = 15;
+    const int autoAF3B14EE_ai = 1;
+
+    // Variable Initialization
+
+    // Implementation
+    lv_player_index = 2;
+    for ( ; ( (autoAF3B14EE_ai >= 0 && lv_player_index <= autoAF3B14EE_ae) || (autoAF3B14EE_ai < 0 && lv_player_index >= autoAF3B14EE_ae) ) ; lv_player_index += autoAF3B14EE_ai ) {
+        if (((libNtve_gf_PlayerIsEnemy(1, lv_player_index, libNtve_ge_PlayerRelation_Enemy) == true) || (libNtve_gf_PlayerIsEnemy(1, lv_player_index, libNtve_ge_PlayerRelation_EnemyMutual) == true))) {
+            libNtve_gf_SetUpgradeLevelForPlayer(lv_player_index, "AP_DetectionEnemy", 1);
+        }
+

```

Comment:
Unhardcode P1 from there. Instead use check if the controller is User:
![Snímek obrazovky pořízený 2024-05-16 15-40-04](https://github.com/Ziktofel/Archipelago-SC2-data/assets/3447808/c2eec3ad-c03d-46a6-b13e-fc4226d5d50e)
![Snímek obrazovky pořízený 2024-05-16 15-41-08](https://github.com/Ziktofel/Archipelago-SC2-data/assets/3447808/768b07fd-f7f0-451c-b6bb-6afe7e8fbf5c)

Reply:
I verified some time ago that the human player is player 1 in every campaign mission. If I'm reading the description of this function right, this would involve looping over every player and checking if it's human-controlled. Is it worth the extra visual clutter / for loop for it? Does AP Core have an AP function for this yet?

Reply:
I don't really have a lot of time today, so I just split the human player ID to be gotten from a function (which for now always returns 1). I figure if triggers always draw from there, we can update the function later if we someday add a map where the human player isn't player 1. If you consider it better to get that done in this PR, I can maybe look into it tomorrow or sometime on the weekend.

![image](https://github.com/Ziktofel/Archipelago-SC2-data/assets/31861583/04b267b3-d978-498c-b7aa-8e7669e2aa74)
![image](https://github.com/Ziktofel/Archipelago-SC2-data/assets/31861583/a51faea8-22d2-497c-a32e-9c21282e4b77)

Reply:
I think that better job is to check if the player is human. This mod's scope is just library of units/upgrades and the human player is always unhardcoded (Vanilla scripts are using P1 but in this mod it's switched into a parameter). This mod is written in a way that can still used for example as a dependency for melee-like map (one of the reasons why it's split from triggers mod).  
What you actually did is to hardcode the P1 into another location.

I'd just loop over the players and check the controller (with the function) and if it's the user, do the rest

Reply:
Done
![image](https://github.com/Ziktofel/Archipelago-SC2-data/assets/31861583/8bbfea0b-b3b5-49db-aec9-80b226568c9d)

Did a quick test on Amon's Reach and the detection circles still showed up.